jazkarta/shop/interfaces.py

```diff
@@ -65,10 +72,21 @@ class IProduct(model.Schema):
         required=False,
     )
 
+    recommended_products = RelationList(
+        title='Recommended products',
+        description=u'Recommendations to users who bought this product, shown during checkout.',
+        default=[],
+        required=False,
+        value_type=schema.Choice(
+            source=CatalogSource(
+                object_provides='jazkarta.shop.interfaces.IProduct'),
```

Comment:
I'm finding that this vocabulary restriction option makes it nearly impossible to find book products to select on SDHSP. I am able to find and select journal issues though. I think using the full vocabulary with the selection restriction in the widget options will work better here, I'll try a couple things out.

Reply:
There's no good way to filter the selectable content in the widget except via portal_type, which won't work here, and filtering the result set doesn't work too well. I think the fix for this has to be on the SDHSP side ensuring that the book productions are indexed in a way that makes them easier to find.